app/controllers/three_d_secure_controller.js

```diff
@@ -98,16 +98,26 @@ module.exports = {
     const md = _.get(charge, 'auth3dsData.md')
     const htmlOut = _.get(charge, 'auth3dsData.htmlOut')
     const worldpayChallengeJwt = _.get(charge, 'auth3dsData.worldpayChallengeJwt')
+    const threeDSReturnUrl = `${req.protocol}://${req.hostname}${paths.generateRoute('external.card.auth3dsRequiredIn', { chargeId: charge.id })}`
 
     if (issuerUrl && paRequest) {
       const data = {
         postUrl: issuerUrl,
         paRequest: paRequest,
-        threeDSReturnUrl: `${req.protocol}://${req.hostname}${paths.generateRoute('external.card.auth3dsRequiredIn', { chargeId: charge.id })}`
+        threeDSReturnUrl: threeDSReturnUrl
       }
       if (md) {
         data.md = md
       }
+
+      logger.info('Form details sent to card issuer', {
+        ...getLoggingFields(req),
+        post_url: issuerUrl,
+        pa_request: paRequest,
```

Comment:
I think we probably want to use the actual names of the form fields rather than the names of the variables we put in them (so `PaReq` rather than `paRequest`) in order to make it easier to reconstruct what will actually be posted.

There’s also the issue of that the form fields will get jumbled up with all of the other fields we log. I haven’t actually tried this but I assume we could maybe use nested objects to make this clear. Something like this:

```js
logger.info('Rendering form to posts to card issuer or gateway to initiate 3DS authentication',
  ...getLoggingFields(req),
  form: {
    action: issuerUrl,
    fields: {
      PaReq: paRequest,
      MD: md,
      TermUrl: threeDSReturnUrl
    }
  }
```

Might be worth trying that and seeing what happens.

Reply:
Done